cookbook/rule/general_information_on_rule_format.rst

```diff
@@ -0,0 +1,543 @@
+
+Quick Overview
+--------------
+
+**This cookbook is about a feature only provided in the Enterprise Edition.**
+
+Enrichment rules enable calculating the value of an attribute according to specific conditions. Those rules are regularly
+applied, as a user who would regularly edit product.
+
+File Structure
+--------------
+
+Enrichment rules are defined in YAML. The file extension has to be ".yml". Indentation is mandatory within the
+file and has to follow the YAML format strictly. To use a rule it has to be imported in the PIM.
+
+This file starts with "rules" root element, which contains the list of enrichment rules. This document is about this
+list. Each rule is referred to by a code and can contain a list of conditions and actions.
+
+.. code-block:: yaml
+
+    # Example of a file with 2 rules
+    rules:
+        camera_set_canon_brand:
+            priority: 0
+            conditions:
+                - field: family.code
+                  operator: IN
+                  value:
+                    - camcorders
+                - field: name
+                  operator: CONTAINS
+                  value: Canon
+            actions:
+                - type: set_value
+                  field: camera_brand
+                  value: canon_brand
+        camera_copy_name_to_model:
+            priority: 0
+            conditions:
+                - field: family.code
+                  operator: IN
+                  value:
+                    - camcorders
+            actions:
+                - type: copy_value
+                  from_field: name
+                  to_field: camera_model_name
+
+Indentation is mandatory within the file and must be strictly identical to the one shown in the example.
+
+.. tip::
+
+    To apply a rule on every product, set the following condition:
+
+    .. code-block:: yaml
+
+        conditions:[]
+        actions:
+          - type:
+
+Enrichment Rule Structure
+-------------------------
+
+Structure’s elements which define a rule are :
+ - rule code (dynamic)
+ - priority*
+ - conditions
+ - actions
+
+Structure's elements which define a condition are :
+ - field
+ - locale​*
+ - scope​*
+ - operator
+ - value
+
+An enrichment rule is structured as follows :
+
+.. code-block:: yaml
+
+    [free rule code]:
+        priority​*:
+        conditions:
+            - field:
+              locale​*:
+              scope​*:
+              operator:
+              value:
+        actions:
+            - type:
+              [Diverse elements according to the action]
+
+Elements with * are optionals.
+
+**Dashes** - ​before element field and after each element contained in value part are mandatory.
+
+**Colon** : ​mandatory after each structure element.
+
+.. tip::
+
+    For more details you can see the `YAML specifications <http://yaml.org/spec/>`_.
+
+.. warning::
+
+    Rules code choice is up to you, however it has to contain only alphanumeric characters, underscores and dashes and 255
```

Comment:
only alphanumeric characters, underscores, dashes and be less than 255 characters